// [[Reason: checker.typeToTypeNode deliberately fails on types that originate from node_modules.]] ////

//// [tests/cases/compiler/declarationEmitReexportedSymlinkReference3.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -2,21 +2,27 @@
 
 //// [/.src/monorepo/pkg3/dist/index.d.ts]
 export * from './keys';
 //# sourceMappingURL=index.d.ts.map
+//// [/.src/monorepo/pkg3/dist/keys.d.ts]
+export declare const ADMIN: invalid;
+//# sourceMappingURL=keys.d.ts.map
 /// [Errors] ////
 
 monorepo/pkg3/src/keys.ts(3,14): error TS2742: The inferred type of 'ADMIN' cannot be named without a reference to '../../pkg2/node_modules/@raymondfeng/pkg1/dist'. This is likely not portable. A type annotation is necessary.
+monorepo/pkg3/src/keys.ts(3,22): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 
 
 ==== monorepo/pkg3/src/index.ts (0 errors) ====
     export * from './keys';
-==== monorepo/pkg3/src/keys.ts (1 errors) ====
+==== monorepo/pkg3/src/keys.ts (2 errors) ====
     import {MetadataAccessor} from "@raymondfeng/pkg2";
     
     export const ADMIN = MetadataAccessor.create<boolean>('1');
                  ~~~~~
 !!! error TS2742: The inferred type of 'ADMIN' cannot be named without a reference to '../../pkg2/node_modules/@raymondfeng/pkg1/dist'. This is likely not portable. A type annotation is necessary.
+                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 ==== monorepo/pkg1/dist/index.d.ts (0 errors) ====
     export * from './types';
 ==== monorepo/pkg1/dist/types.d.ts (0 errors) ====
     export declare type A = {
